spring-cloud-skipper-docs/src/main/asciidoc/security.adoc

```diff
@@ -0,0 +1,337 @@
+====
+
+[[configuration-security-oauth2]]
+== OAuth 2.0 Security
+
+https://oauth.net/2/[OAuth 2.0] allows you to integrate Spring Cloud
+Skipper into Single Sign On (SSO) environments. The following 3 OAuth2 Grant Types will be used:
+
+* _Password_ - Used by the shell (And the REST integration), so you can login using username and password
+* _Client Credentials_ - Retrieve an Access Token directly from your OAuth provider and pass it to the Skipper server using the _Authorization_ Http header.
+
+The REST endpoints can be accessed using 2 ways:
+
+* Basic Authentication which will use the _Password Grant Type_ under the covers to authenticate with your OAuth2 service
+* Access Token which will use the _Client Credentials Grant Type_ under the covers
+
+NOTE: When authentication is set up, it is strongly recommended to enable HTTPS
+as well, especially in production environments.
+
+You can turn on OAuth2 authentication by adding the following to `skipper.yml` or via
+environment variables:
+
+[source,yaml]
+----
+security:
+  oauth2:
+    client:
+      client-id: myclient                                             # <1>
+      client-secret: mysecret
+      access-token-uri: http://127.0.0.1:9999/oauth/token
+      user-authorization-uri: http://127.0.0.1:9999/oauth/authorize
+    resource:
+      user-info-uri: http://127.0.0.1:9999/me
+----
+
+<1> Providing the Client Id in the OAuth Configuration Section will activate OAuth2 security
+
+You can verify that basic authentication is working properly using _curl_:
+
+[source,bash]
+----
+$ curl -u myusername:mypassword http://localhost:7577/
+----
+
+As a result you should see a list of available REST endpoints.
+
+Besides Basic Authentication, you can also provide an Access Token in order to
+access the REST Api. In order to make that happen, you would retrieve an
+OAuth2 Access Token from your OAuth2 provider first, and then pass that Access Token to
+the REST Api using the **Authorization** Http header:
+
+```
+$ curl -H "Authorization: Bearer <ACCESS_TOKEN>" http://localhost:7577/
+```
+
+[[configuration-security-oauth2-authorization]]
+==== OAuth REST Endpoint Authorization
+
+Spring Cloud Skipper supports the following 3 roles:
+
+* *VIEW* for anything that relates to retrieving state
+* *CREATE* for anything that involves creating, deleting or mutating the state of the system
+* *MANAGE* for boot management endpoints.
+
+The rules regarding which REST endpoints require which roles, are specified in
+`application.yml` of the `spring-cloud-skipper-server-core` module.
+
+Nonetheless, you can override those, if desired. The configuration takes
+the form of a YAML *list* (as some rules may have precedence over others) and so
+you'll need to copy/paste the whole list and tailor it to your needs (as there is
+no way to merge lists). Always refer to your version of `application.yml`, as the snippet reproduced below may be out-dated. The default rules are as such:
+
+[source,yaml]
+----
```

Comment:
I guess we wanted to show what's in the application.yml here. I can copy them over here.